### PR TITLE
circle intersect

### PR DESCRIPTION
**Pull request by hgloeckner**
function to get the intersection points of two circles. I haven't written any tests yet, so I am not sure if it is sufficient like this...

### CODE REVIEW

File: orcestra/flightplan.py [hunk 1]
```diff
@@ -485,6 +485,46 @@ def is_entry_through(self):
...
+
+            def cost(fwd_az):
+                fwd_az = (fwd_az + 180) % 360 - 180
+                # print(fwd_az)
+                fwd_res = (
+                    geod.fwd(self.center.lon, self.center.lat, fwd_az, self.radius),
+                )
+                _, _, dist = geod.inv(
+                    fwd_res[0][0], fwd_res[0][1], other.center.lon, other.center.lat
+                )
```
Comment (d70-t) [inline orcestra/flightplan.py:506]: I'd suggest to destructure the result of `geod.fwd` as you do for `geod.inv` to improve readability:
```suggestion
                candidate_lon, candidate_lat, _ = (
                    geod.fwd(self.center.lon, self.center.lat, fwd_az, self.radius),
                )
                _, _, dist = geod.inv(
                    candidate_lon, candidate_lat, other.center.lon, other.center.lat
                )
```
The `candidate_lon`, `candidate_lat` result should be scalar. If not, then maybe `fwd_az ` isn't scalar (but maybe should?)

File: orcestra/flightplan.py [hunk 2]
```diff
@@ -485,6 +485,46 @@ def is_entry_through(self):
...
+                return np.abs(dist - other.radius)
```
Comment (d70-t) [inline orcestra/flightplan.py:507]: Not sure if that actually improves things, but somewhere back in my math classes I've learned that quadratic cost functions tend to have better convergence. If that's true, then we might want something like:

```suggestion
                return (dist - other.radius)**2
```

**Review by d70-t — COMMENTED** [review]
I've put some minor comments.

Thanks for the very nice addition 👍 🎉

File: orcestra/flightplan.py (hunk 2 above)
Comment (hgloeckner) [reply orcestra/flightplan.py]: I changed this  - in order for it to work with the square, I had to change to the default minimize method, but that's no problem I guess.

File: orcestra/flightplan.py (hunk 1 above)
Comment (hgloeckner) [reply orcestra/flightplan.py]: yes so true. changed it

**Review by d70-t — APPROVED** [review]
Thanks a lot :+1:

**Pull request merged**